chess-AI/nn_layout.py

```diff
@@ -23,116 +23,94 @@ class PlayNetwork(nn.Module):
                                                   bias=False),
                                         nn.BatchNorm2d(1),
-                                        nn.ReLU(inplace=True),
+                                        nn.ReLU(),
                                         nn.Flatten(),
                                         nn.Linear(in_features=8 * 8,
-                                                  out_features=256),
-                                        nn.ReLU(inplace = True),
-                                        nn.Linear(in_features=256,
+                                                  out_features=self.num_filters),
+                                        nn.ReLU(),
+                                        nn.Linear(in_features=self.num_filters,
                                                   out_features=1),
                                         nn.Tanh())
 
-        
     def forward(self, x):
         x = self.conv_layer(x)
         
         # Go through all the residual blocks and add the
         # input of the block to the output before regularization.
-        for num_res in range(self.num_res_blocks):
+        for res_block in self.res_blocks:
             shortcut = x
-            x = self.res_block(x)
+            x = res_block(x)
             x = nn.functional.relu_(x + shortcut)
         
         value_out = self.value_head(x)
         policy_out = self.policy_head(x)
         
-        return (policy_out, value_out)
-
-
-    def predict(self, board, input):
-        """Function to get neural network output.
-        """
-        # # Note: uncomment below two lines and comment out the other code to run MCTS without neural network.
-        # policy = np.arange(8 * 8 * 73).reshape(8, 8, 73)
-        # value = 1
-
-        model = PlayNetwork()
-        policy, value = model(torch.randn(1, 119, 8, 8))
-        value = value.item()
-
-        policy = policy.reshape(8, 8, 73)
+        policy = policy_out.reshape(8, 8, 73)
+        return policy, value_out
 
-        # TODO: this conversion should be done at the calling function, predict() should return
-        # the policy and value. Then, when you have the policy, you can convert it. This way we
-        # don't need to pass board to this function either. This should solely take the input
-        # state, run it through the net, then return the output.
-        policy_converter = PlayNetworkPolicyConverter()
-        move_values = policy_converter.find_value_of_all_legal_moves(policy, board)
 
-        return value, move_values
-                
 def main():
-    model = PlayNetwork()
-    policy, value = model(torch.randn(1, 119, 8, 8))
-    policy = policy.reshape(8, 8, 73)
-    value = value.item()
-    print(policy)
-    print(value)
+    model = PlayNetwork().float()
 
     import chess
     board = chess.Board()
     policy_converter = PlayNetworkPolicyConverter()
 
+    policy, value = model(get_cnn_input(board).float())
```

Comment:
Same comment as above about if we need this to be `.float()` let's have that happen in `get_cnn_input()`